lib/carelink/bolusNWizard.js

```diff
@@ -470,129 +472,164 @@ module.exports = function(timezone, model) {
   }
 
   var datumsBySeqNum = null, findNormal = null, findSquare = null, findWizard = null;
+  var pointer = 0, cachedObjects = null;
 
-  return function(simulator, datum, i, data) {
-    var normal = null, square = null, wizard = null;
-    var parsed = parser(datum);
+  function connectEvents(data) {
+    cachedObjects = [];
+    for (var i = 0; i < data.length; ++i) {
+      var normal = null, square = null, wizard = null;
+      var parsed = parser(data[i]);
 
-    if (datumsBySeqNum == null) {
-      buildCache(data);
-    }
+      if (datumsBySeqNum == null) {
+        buildCache(data);
+      }
 
-    if (findNormal == null) {
-      findNormal = makeFindNormalFn();
-    }
+      if (findNormal == null) {
+        findNormal = makeFindNormalFn();
+      }
 
-    if (findSquare == null) {
-      findSquare = makeFindSquareFn();
-    }
+      if (findSquare == null) {
+        findSquare = makeFindSquareFn();
+      }
 
-    if (findWizard == null) {
-      findWizard = makeFindWizardFn();
-    }
-    
-    if (parsed != null) {
-      switch(parsed.type) {
-        case 'bolusNormal':
-          if (isInCache(parsed)) {
-            removeFromCache(parsed);
-            normal = cleanEvent(parsed);
-            if (parsed.dualComponent) {
-              square = findSquare(parsed);
-              if (square != null) {
-                datumsBySeqNum[square.uploadId][square.uploadSeqNum] = null;
+      if (findWizard == null) {
+        findWizard = makeFindWizardFn();
+      }
+      
+      if (parsed != null) {
+        switch(parsed.type) {
+          case 'bolusNormal':
+            if (isInCache(parsed)) {
```

Comment:
In every one of these cases (and yes, I realize this is old code and it's just a layer deeper) you ask if it's in the cache...and then do nothing if it isn't in the cache. Shouldn't that be an error condition?

Similarly with the test for parsed -- what happens if parsed IS null? If that can't reasonably occur, why the test? If it can occur, is it correct to just ignore it?